Solution1/BusinessLayer/Services/WeatherService.cs

```diff
@@ -127,5 +134,25 @@ public async Task<ForecastWeatherDTO> GetForecastByCityNameAsync(string cityName
                             .ToDictionary(k => k.Key, v => v.Select(response => response));
             return result;
         }
+
+        public async Task BackgroundSaveWeatherAsync(IEnumerable<string> cities)
+        {
+            try
+            {
+                var weathers = cities.Select(async city =>
```

Comment:
mapper can map all collection at once, not necessary to use select, and you have separate method for getting weather by array of cities

Reply:
+